### PR TITLE
feat(cb2-6757): test records flow submit

### PR DESCRIPTION
**Pull request by Ruairidh-BJSS**
Add check answers page to view the test result before submitting

https://dvsa.atlassian.net/browse/CB2-6757

## Checklist

- [ ] Branch is rebased against the latest develop/common
- [ ] Necessary `id` required prepended with `"test-"` have been checked with automation testers and added
- [ ] Code and UI has been tested manually after the additional changes
- [ ] PR title includes the JIRA ticket number
- [ ] Squashed commits contain the JIRA ticket number
- [ ] Link to the PR added to the repo
- [ ] Delete branch after merge

### CODE REVIEW

**Review by mcadenas-bjss — DISMISSED** [review]

**Review by kpearmanbjss — DISMISSED** [review]

File: src/app/features/test-records/components/base-test-record/base-test-record.component.ts [hunk 1]
```diff
@@ -27,6 +27,7 @@ export class BaseTestRecordComponent implements AfterViewInit {
   @Input() testResult!: TestResultModel;
   @Input() isEditing: boolean = false;
   @Input() expandSections = false;
+  @Input() isContingency = false;
```
Comment (andrewf-bjss) [inline src/app/features/test-records/components/base-test-record/base-test-record.component.ts:30]: Could we show this same review page for Test Result amendment as well?  (i.e. not just Contingency tests?)

**Review by andrewf-bjss — CHANGES_REQUESTED** [review]
Could we show this same review page for Test Result amendment as well? (i.e. not just Contingency tests?)

**Review by kpearmanbjss — DISMISSED** [review]

**Review by mcadenas-bjss — DISMISSED** [review]

**Review by andrewf-bjss — APPROVED** [review]
Thank you  🙂

File: src/app/features/test-records/create/views/create-test-record/create-test-record.component.ts [hunk 2]
```diff
@@ -87,6 +88,17 @@ export class CreateTestRecordComponent implements OnInit, OnDestroy, AfterViewIn
     this.testRecordsService.createTestResult(cloneDeep(testResult));
   }
 
+  async handleReview(): Promise<void> {
+    if (this.isAnyFormInvalid()) {
+      return;
+    }
+    this.testMode = TestModeEnum.View;
+  }
```
Comment (gjulien-bjss) [inline src/app/features/test-records/create/views/create-test-record/create-test-record.component.ts:96]: why is this async?

**Comment by Ruairidh-BJSS** [comment]
Not to be merged until after VTM cut

**Review by mcadenas-bjss — DISMISSED** [review]

File: src/app/features/test-records/amend/views/test-record/test-record.component.ts [hunk 3]
```diff
@@ -111,16 +130,16 @@ export class TestRecordComponent implements OnInit, OnDestroy {
...
-
-    this.testRecordsService.updateTestResult(cloneDeep(testResult));
+    return false;
   }
 
   handleCancel() {
-    this.isCanceled = !this.isCanceled;
+    this.testMode === TestModeEnum.Cancel || this.testMode === TestModeEnum.View
+      ? (this.testMode = TestModeEnum.Edit)
+      : (this.testMode = TestModeEnum.Cancel);
```
Comment (tahayazici) [inline src/app/features/test-records/amend/views/test-record/test-record.component.ts:142]:
```suggestion
    this.testMode = this.testMode === TestModeEnum.Cancel || this.testMode === TestModeEnum.View
      ? TestModeEnum.Edit
      : TestModeEnum.Cancel;
```

File: src/app/features/test-records/create/views/create-test-record/create-test-record.component.html [hunk 4]
```diff
@@ -1,25 +1,43 @@
 <ng-container *ngIf="testResult$ | async as testResult">
-  <ng-container *ngIf="!isAbandon; else abandoned">
-    <app-base-test-record
-      [isEditing]="true"
-      [testResult]="testResult"
-      [expandSections]="true"
-      (newTestResult)="handleNewTestResult($event)"
-    ></app-base-test-record>
+  <ng-container [ngSwitch]="testMode">
```
Comment (tahayazici) [inline src/app/features/test-records/create/views/create-test-record/create-test-record.component.html:2]: You can combine this and the above `<ng-container>`
```suggestion
<ng-container *ngIf="testResult$ | async as testResult" [ngSwitch]="testMode">
```

**Review by mcadenas-bjss — APPROVED** [review]

**Review by tahayazici — APPROVED** [review]

**Pull request merged**